internal/nhctl/controller/dev_start.go

```diff
@@ -561,7 +568,7 @@ func containerStatusForDevPod(pod *corev1.Pod, consumeFun func(status string, er
 }
 
 func (c *Controller) genContainersAndVolumes(podSpec *corev1.PodSpec,
-	containerName, devImage, storageClass string, duplicateDevMode bool) (*corev1.Container,
+	containerName, devImage, storageClass string, duplicateDevMode bool, sshUsed bool) (*corev1.Container,
```

Comment:
You cat get all `ContainerDevConfig` from `*Controller`.  
No need to add `genContainersAndVolumes()` input parameter `sshUsed bool`